srcCpp03/PerftestTransport.cxx

```diff
@@ -332,42 +415,160 @@ void configureShmemTransport(
+     * the message_size_max - RTPS headers.
+     */
     int fragmentSize = (std::min)(
             parentMsgSizeMax - COMMEND_WRITER_MAX_RTPS_OVERHEAD,
-            flowControllerTokenSize);
+            flowControllerTokenSize - COMMEND_WRITER_MAX_RTPS_OVERHEAD);
 
-    // max(1, (sample_serialized_size / fragment_size))
-    unsigned long long int rtpsMessagesPerSample = (std::max)(
-            1ull, (perftest_cpp::OVERHEAD_BYTES + datalen) / fragmentSize);
+    unsigned long long rtpsMessagesPerSample = (std::max)(
+            1ull, (datalen / fragmentSize) + 1);
 
-    unsigned long long int receivedMessageCountMax =
+    unsigned long long receivedMessageCountMax =
             2 * (_PM->get<int>("sendQueueSize") + 1) * rtpsMessagesPerSample;
 
-    // min(maxBufferSize, received_message_count_max * rtps_message_size)
-    unsigned long long int receiveBufferSize = (std::min)(
+    unsigned long long receiveBufferSize = (std::min)(
         (unsigned long long) maxBufferSize,
         receivedMessageCountMax *
                 (COMMEND_WRITER_MAX_RTPS_OVERHEAD + fragmentSize));
 
-    // Avoid bottleneck due to SHMEM.
-    ss << receivedMessageCountMax;
-    qos_properties["dds.transport.shmem.builtin.received_message_count_max"] =
-        ss.str();
+    if (qos_properties["dds.transport.shmem.builtin.received_message_count_max"].empty()) {
+        ss.str("");
+        ss.clear();
+        ss << receivedMessageCountMax;
+        qos_properties["dds.transport.shmem.builtin.received_message_count_max"] = ss.str();
+        transport.loggingString +=
+                ("\tSHMEM received_message_count_max: "
+                + ss.str()
+                + "\n");
+    }
 
-    ss.str("");
-    ss.clear();
-    ss << receiveBufferSize;
-    qos_properties["dds.transport.shmem.builtin.receive_buffer_size"] =
-        ss.str();
+    if (qos_properties["dds.transport.shmem.builtin.receive_buffer_size"].empty()) {
+        ss.str("");
+        ss.clear();
+        ss << receiveBufferSize;
+        qos_properties["dds.transport.shmem.builtin.receive_buffer_size"] = ss.str();
+        transport.loggingString +=
+                ("\tSHMEM receive_buffer_size: "
+                + ss.str()
+                + "\n");
+    }
+}
+
+/*
+ * Gets the MessageSizeMax given the name (String) of a transport from a
+ * DDS_DomainParticipantQos object. If the value is not present, returns
+ * DEFAULT_MESSAGE_SIZE_MAX.
+ */
+long getTransportMessageSizeMax(
+        std::string targetTransportName,
+        PerftestTransport &transport,
+        std::map<std::string, std::string> &qos_properties)
+{
+    std::string propertyName =
+            transport.transportConfigMap[targetTransportName].prefixString
+            + ".parent.message_size_max";
+
+    if (qos_properties.find(propertyName) != qos_properties.end()) {
+        //printf("Value for %s is %s\n", propertyName.c_str(), qos_properties[propertyName].c_str());
```

Comment:
Why this is commented?

Reply:
I guess it can be removed, it was a debug statement and I found it useful in case we ever have more debug levels.